api/provider/aws/logging.go

```diff
@@ -16,6 +16,73 @@ const (
 	MAX_DESCRIBE_STREAMS_COUNT = 1000
 )
 
+func GetLogsFromCloudTrail(
+	cloudWatchLogsAPI cloudwatchlogsiface.CloudWatchLogsAPI,
+	logGroupName string,
+	tail int,
+	start,
+	end time.Time,
+	filterPattern string,
+) ([]models.LogFile, error) {
+	logStreams, err := describeLogStreams(cloudWatchLogsAPI, logGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	logFiles := []models.LogFile{}
+	for _, logStream := range logStreams {
+		if strings.Contains(aws.StringValue(logStream.LogStreamName), "_CloudTrail_") {
+			logStreamName := aws.StringValue(logStream.LogStreamName)
+			logStreamNames := []string{logStreamName}
+
+			input := &cloudwatchlogs.FilterLogEventsInput{}
+			input.SetLogGroupName(logGroupName)
+			input.SetLogStreamNames(aws.StringSlice(logStreamNames))
+			input.SetFilterPattern(filterPattern)
+
+			if tail != 0 {
+				input.SetLimit(int64(tail))
+			}
+
+			if !start.IsZero() {
+				startMS := timeToMilliseconds(start)
+				input.SetStartTime(startMS)
+			}
+
+			if !end.IsZero() {
+				endMS := timeToMilliseconds(end)
+				input.SetEndTime(endMS)
+			}
+
+			var previousToken string
```

Comment:
I would expect `output.NextToken` to be nil and the lastPage to be true if there are no further results. Wouldn't checking for this condition be sufficient so you don't have to keep track of the previousToken?

For example:
```go
eventsFN := func(...) {
    // append code

    return output.NextToken != nil or return lastPage
}
```